recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,218 @@
     }
 
     default_options = {
-        "boost_optional": False,
-        "boost_variant": False,
-        "coroutines": False,
+        "with_boost": False,
+        "no_std_coroutines": True,          #TODO: how to make checks similar to what are made in Cmake https://github.com/stlab/libraries/blob/main/cmake/StlabUtil.cmake#L35
+        "future_coroutines": False,
         "task_system": "auto",
+        "thread_system": "auto",
     }
 
-    no_copy_source = True
-    _source_subfolder = 'source_subfolder'
+    short_paths = True
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
 
-    def _use_boost(self):
-        return self.options.boost_optional or self.options.boost_variant
+    @property
+    def _compilers_minimum_version(self):
+        return {"gcc": "9",
+                "Visual Studio": "15.8",
+                "clang": "8",
+                "apple-clang": "13",
+                }
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def _requires_libdispatch(self):
-        # On macOS it is not necessary to use the libdispatch conan package, because the library is 
+        # On macOS it is not necessary to use the libdispatch conan package, because the library is
         # included in the OS.
         return self.options.task_system == "libdispatch" and self.settings.os != "Macos"
 
+    def build_requirements(self):
+        self.build_requires("cmake/3.23.3")
+
     def requirements(self):
-        if self._use_boost():
-            self.requires("boost/1.75.0")
+        if self.options.with_boost:
+            self.requires("boost/1.80.0")
 
         if self._requires_libdispatch():
             self.requires("libdispatch/5.3.2")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "libraries-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
-
-    def _fix_boost_components(self):
-        if self.settings.os != "Macos": return
-        if self.settings.compiler != "apple-clang": return
-        if Version(self.settings.compiler.version) >= "12": return
-
-        #
-        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 is not complete.
-        #
-        self.output.info("Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead.")
-        self.options.boost_optional = True
-        self.options.boost_variant = True
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def _default_task_system(self):
-        if self.settings.os == "Macos": 
+        if self.settings.os == "Macos":
             return "libdispatch"
 
         if self.settings.os == "Windows":
             return "windows"
 
-        if self.settings.os == "Emscripten":
-            return "emscripten"
```

Comment:
This was kept else where https://github.com/conan-io/conan-center-index/pull/13048/files#diff-c4747a2c3a2f1b6ca9f4b271de4f2d8958e72c24e71f92a4c3a8bc96f460c385R107-R108 🤔 

Is there a reason this is removed?

Reply:
Most of the changed in the new Conan recipe (`all` directory) reflect changes in the original project.
They changed a lot of things.

In this case:

https://github.com/stlab/libraries/blob/main/cmake/StlabUtil.cmake#L122